xtext-website/documentation/103_domainmodelnextsteps.md

```diff
@@ -235,46 +285,59 @@ This tutorial is about testing the parser and the linker for the *Domainmodel*.
-
-Locate the class *DomainmodelValidator* in the package *org.example.domainmodel.validation* of the language project. Defining the constraint itself is only a matter of a few lines of code:
-
-```java
-@Check
-public void checkNameStartsWithCapital(Entity entity) {
-    if (!Character.isUpperCase(entity.getName().charAt(0))) {
-        warning("Name should start with a capital", 
-            DomainmodelPackage.Literals.TYPE__NAME);
-    }
-}
-```
-
-Any name for the method will do. The important thing is the [Check]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/validation/Check.java) annotation that advises the framework to use the method as a validation rule. If the name starts with a lower case letter, a warning will be attached to the name of the *Entity*.
-
-The second validation rule is straight-forward, too. We traverse the inheritance hierarchy of the *Entity* and look for features with equal names.
+    ```xtend
+	@Inject extension ParseHelper<Domainmodel>
+
+	@Inject extension ValidationTestHelper
+
+	@Test
+	def testNameStartsWithCapitalWarning() {
+	    "entity myEntity {
+	        parent: myEntity
+	    }".parse.assertWarning(
+        DomainmodelPackage.Literals.ENTITY
+        DomainmodelValidator.INVALID_NAME)
+	}
+
+	@Test
+	def testValidModel() {
+	    "entity MyEntity {
+		parent: MyEntity
+	    }".parse.assertNoIssues
+	}
+    ```
 
-```java
-@Check
-public void checkFeatureNameIsUnique(Feature f) {
-    Entity superEntity = ((Entity) f.eContainer()).getSuperType();
-    while (superEntity != null) {
-        for (Feature other : superEntity.getFeatures()) {
-            if (Objects.equal(f.getName(), other.getName())) {
-                error("Feature names have to be unique", DomainmodelPackage.Literals.FEATURE__NAME);
-                return;
-            }
-        }
-        superEntity = superEntity.getSuperType();
-    }
-}
-```
+3.  After saving the Xtend file, it is time to run the test. Select *Run As &rarr; JUnit Test* from the editor's context menu.
 
-The sibling features that are defined in the same entity are automatically validated by the Xtext framework, thus they do not have to be checked twice. Note that this implementation is not optimal in terms of execution time because the iteration over the features is done for all features of each entity.
+These tests serve only as a starting point and can be extended to cover the different features of the language. You can find more examples in the example projects shipped with the Xtext Framework. Simply go to *File &rarr; New &rarr; Example &rarr; Xtext Examples*.
```

Comment:
I think it would be a good idea to include something like: 
`.. different features of the language. **As a small exercise, you could implement e.g. test cases for the checkFeatureNameIsUnique validation rule.** You can find more examples...`